src/containers/DataPackageViewContainer.js

```diff
@@ -51,28 +79,47 @@ export class DataPackageViewContainer extends React.Component {
   }
 
   async componentWillReceiveProps(nextProps) {
+    let graphType
     if(nextProps.datapackage.resources) {
       //check if resources are received by comparing descriptor's resources and
       //received data length
       if(nextProps.datapackage.resources.length == nextProps.resources[0].length) {
         if(nextProps.datapackage.views) {
-          let data = await this.convertData(
-            nextProps.resources[0][this.props.idx],
-            nextProps.datapackage
-          )
-          let layout = await this.generateSpec(nextProps.datapackage.views[this.props.idx])
-          this.setState({
-            data: data,
-            layout: layout
-          })
-
+          graphType = nextProps.datapackage.views[this.props.idx].type
         }
       }
     }
+
+    if(graphType == "vega-lite") {
+      let vlSpec = await this.generateVegaLiteSpec(
+        nextProps.resources[0][this.props.idx],
+        nextProps.datapackage.views[this.props.idx]
+      )
+      this.setState({
+        graphType: graphType,
+        vlSpec: vlSpec
+      })
+    } else {
+      let data = await this.convertData(
+        nextProps.resources[0][this.props.idx],
+        nextProps.datapackage
+      )
+      let layout = await this.generateSpec(nextProps.datapackage.views[this.props.idx])
+      this.setState({
+        graphType: graphType,
+        data: data,
+        layout: layout
+      })
+    }
   }
 
   render() {
     //render PlotlyChart component and pass data, layout, and index
```

Comment:
Can you please also fix the description of DataPackageViewContainer in the topmost comment?